src/js/structs/ServicesList.js

```diff
@@ -25,6 +25,14 @@ class ServicesList extends List {
           });
         });
       }
+
+      if (filters.status != null && filters.status.length !== 0) {
+        services = services.filter(function (service) {
+          return filters.status.some(function (statusValue) {
+            return service.getServiceStatus().key === parseInt(statusValue, 10);
+          });
+        });
+      }
```

Comment:
@pierlo-upitup No. Sorry, I wasn't clear enough. I was referring to [`Tree.prototype.filterItems`](https://github.com/dcos/dcos-ui/blob/f156ca9b6449f39dbaf6687a951815ad53a21639/src/js/structs/Tree.js#L53) which is the method that is used to implement the [`ServiceTree.prototype.findItemById`](https://github.com/dcos/dcos-ui/blob/f156ca9b6449f39dbaf6687a951815ad53a21639/src/js/structs/ServiceTree.js#L98) as well as `filterItemsByText` . But I wouldn't introduce any new methods on the tree and rather use `filterItems` directly as the filter behavior is very specific to the service table and not necessarily part of the data structure.